scripts/amx/README.md

```diff
@@ -0,0 +1,116 @@
+ 
+- **AMX-TILE:** The tile architecture foundation. <br />
+- **AMX-BF16:** Support for BFloat16 operations. <br />
+- **AMX-INT8:** Support for 8-bit integer operations.
+
+### Why Check for AMX Support?
+AMX can provide significant performance improvements for:
+
+- Machine learning applications
+- AI inference workloads
+- Scientific computing
+- Applications using matrix operations
+
+Detecting AMX support helps determine if your system can benefit from AMX-optimized software and libraries like Intel's oneAPI, PyTorch with Intel extensions, or TensorFlow with oneDNN.
+
+## Prerequistes
+### Linux
+- Python 3.6 or higher
+- cpuid utility installed
+```bash
+sudo apt-get update
+sudo apt-get install cpuid
+```
+- To verify that cpuid is installed correctly run:
+```bash
+cpuid -v
+```
+
+### Windows
+- Python 3.6 or higher
+- Microsoft Visual Studio 2022 <br />
+Once Visual Studio is installed click on the Tools tab and navigate to the "Get Tools and Features" section. Ensure that the following settings are selected and then select install while downloading in the bottom right. 
+    <br/>![alt text](image-3.png) 
+
+**Note:** CPUID instruction capability is not available on windows. This instruction will be called from an executable file instead when running on windows.
+
+## Usage
+### Linux 
+1. Launch an instance on desired platform and SSH into instance
+2. Create AMX detection file 
+```bash
+nano amx-detect-linux.py
+```
+3. Paste contents of the amx-detect-linux.py file into the newly created file and save.
+4. Make the script executable 
+```bash
+chmod +x amx-detect-linux.py 
+```
+5. Run the script
+```bash
+sudo python3 amx-detect-linux.py
+```
+### Expected Output 
+If AMX is enabled on the CPU the script should output the following
+
+![alt text](image-1.png)
+
+**Note:** The following command can be used to query specific CPU feature information:
+```bash
+cpuid -l 7 -s 0
+```
+
+-l 7: Specifies CPUID leaf 7, which contains information about extended features
+-s 0: Specifies sub-leaf 0 of leaf 7
+
+Leaf 7, sub-leaf 0 contains information about various CPU extensions including AVX-512, AMX, and other advanced features. The script parses this output to detect AMX-specific capabilities. This particular leaf/sub-leaf combination is used because Intel placed AMX feature flags in this location of the CPUID instruction output.
+You can manually run this command to see the raw CPUID information that the script parses:
+
+![alt text](image-2.png)
+
+
+
+### Windows
+1. Launch Windows instance and RDP into instance
+2. Download the cpuid_feature_bits.c source file and save to the downloads folder
```

Comment:
2. Git clone repository or download the cpuid_feature_bits.c source file manually to the downloads folder